lib/vpt/trigger/trigger-animation.ts

```diff
@@ -53,10 +53,10 @@ export class TriggerAnimation implements IAnimation {
 		this.unhitEvent = true;
 	}
 
-	public updateAnimation(physics: PlayerPhysics, table: Table) {
-		const oldTimeMsec = (this.timeMsec < physics.timeMsec) ? this.timeMsec : physics.timeMsec;
-		this.timeMsec = physics.timeMsec;
-		const diffTimeMsec = physics.timeMsec - oldTimeMsec;
+	public updateAnimation(timeMsec: number, table: Table) {
+		const oldTimeMsec = (this.timeMsec < timeMsec) ? this.timeMsec : timeMsec;
+		this.timeMsec = timeMsec;
+		const diffTimeMsec = timeMsec - oldTimeMsec;
```

Comment:
4 times the same code - and no, I don't have a better idea for DRY

Reply:
Yeah, because it modifies a member variable. I could move the code into a parent class, but there is none, so I'd have to create one, but for only three lines of code, that would be overkilling it.